### PR TITLE
Cosmos: Support AAD RBAC via TokenCredential

### LINKED ISSUE #25063: How do I get the Cosmos database Id so I can use GetCosmosClient()
## Question

Occasionally EF doesn't support what I need so I want to bypass it and make a quick cosmos query.  I thought I could do this easily by using the `DbContext.GetCosmosClient()` extension method.  However, I've run into the issue that I still need to know the database id and container id in order to get a container to query on.

I want to do something like this...
```
var ctx = ServiceProvider.GetRequiredService<MyCosmosDbContext>();
var client = ctx.Database.GetCosmosClient();
var databaseId = HOW_DO_I_GET_THIS;
var containerName = ctx.Model.GetDefaultContainer();
var container = client.GetContainer(databaseId, containerName);
var result = container.GetItemQueryStreamIterator("query");
```

Of course, somewhere I must call `DbContextOptionsBuilder.UseCosmos(connectionString, databaseId)` but this is different in different environments/applications so I don't have a reliable way to get the database Id that way.

Any help much appreciated.

### Provider and version information

Microsoft.EntityFrameworkCore 5.0.6
Microsoft.EntityFrameworkCore.Cosmos 5.0.6
Microsoft.Azure.Cosmos 3.12.0

### LINKED ISSUE #26491: Cosmos: Support AAD RBAC via the ClientSecretCredential
The Azure Cosmos DB SDK now supports role-based access control (RBAC) via Azure Active Directory (AAD) token credentials.

Instructions: [Configure role-based access control for your Azure Cosmos DB account with Azure  Active Directory](https://docs.microsoft.com/azure/cosmos-db/how-to-setup-rbac#initialize-the-sdk-with-azure-ad)

This more direct support should be recommended over [resource tokens](https://github.com/dotnet/efcore/issues/20969).

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Add a method to get the configured database name

Fixes #26491
Fixes #25063

### CODE REVIEW

**Review by smitpatel — APPROVED** [review]

File: .github/workflows/TestCosmos.yaml [hunk 1]
```diff
@@ -33,3 +33,9 @@ jobs:
       - name: Test on Cosmos
         run: test.cmd /p:Projects=${{ github.workspace }}\test\EFCore.Cosmos.FunctionalTests\EFCore.Cosmos.FunctionalTests.csproj
         shell: cmd
+
+      - name: Publish Test Results
+        uses: EnricoMi/publish-unit-test-result-action/composite@v2
```
Comment (smitpatel) [inline .github/workflows/TestCosmos.yaml:38]: ???

**Comment by codeminic** [comment]
Nice to see that feature! What release will it be in?

**Comment by roji** [comment]
This will most likely be released with the upcoming EF Core 7.0 in November.

**Comment by codeminic** [comment]
@roji thx for the feedback! Will it be in one of the preview versions before rc1?

**Comment by roji** [comment]
@codeminic no, preview7 has already been released, and the next "preview' is rc1 in a month.

**Pull request merged**